### PR TITLE
Add console logging tips

### PR DESCRIPTION
**Pull request by zackhall**
## TL;DR;

Helpful logging techniques using coercion and conditonal breakpoints.
## Username

[@zthall](https://twitter.com/zthall)]
## Using conditional breakpoints to log data

If you wanted to log to the console a value each time a function is called, you can use conditional break points to do this. Open up your dev tools, find the function where you'd like to log data to the console and set a breakpoint with the following condition:

```
console.log(data.value) && false
```

This always evaluates to false, so it will not pause the page thread when it's hit, but it will log data to the console. This can also be used to count how many times a function or callback is called.
## Printing a function variable to console

Have you ever logged a function variable to the console and weren't able to just view the function's code? The quickest way to see the function's code is to coerce it to a string using concatenation with an empty string.

```
console.log(funcVariable + '');
```

### CODE REVIEW

**Comment by kurtextrem** [comment]
:shipit:

**Comment by zenopopovici** [comment]
Simple enough :shipit:

**Comment by drakmail** [comment]
Hm,

> evaluates to false, so it will not pause the page thread when it's hit

Maybe add additional links explaining why it is not pause page thread?

**Comment by zackhall** [comment]
@drakmail I'd be happy to expand on it. Can you help me understand the part you'd like to clarify so that I can make it more clear? :smile: 

A conditional breakpoint pauses the page thread only if the condition for the breakpoint evaluates to true. So by using a condition like `console.log('foo') && false` it's guaranteed to evaluate to false since you're putting the literal false in the AND condition.

Is that helpful?

**Comment by zenopopovici** [comment]
@zackhall Please commit the clarification you suggested :)

**Comment by zackhall** [comment]
np – I updated the pull request with the clarification. :+1:

**Comment by zenopopovici** [comment]
I'm for :shipit: .

File: _posts/en/2016-xx-xx-helpful-console-log-hacks.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+categories:
+    - en
+---
+
+## Using conditional breakpoints to log data
+
+If you wanted to log to the console a value each time a function is called, you can use conditional break points to do this. Open up your dev tools, find the function where you'd like to log data to the console and set a breakpoint with the following condition:
+
+```
```
Comment (loverajoel) [inline _posts/en/2016-xx-xx-helpful-console-log-hacks.md]: add ```js

File: _posts/en/2016-xx-xx-helpful-console-log-hacks.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+console.log(data.value) && false
+```
+
+A conditional breakpoint pauses the page thread only if the condition for the breakpoint evaluates to true. So by using a condition like console.log('foo') && false it's guaranteed to evaluate to false since you're putting the literal false in the AND condition. So this will not pause the page thread when it's hit, but it will log data to the console. This can also be used to count how many times a function or callback is called.
+
+## Printing a function variable to console
+
+Have you ever logged a function variable to the console and weren't able to just view the function's code? The quickest way to see the function's code is to coerce it to a string using concatenation with an empty string.
+
+```
```
Comment (loverajoel) [inline _posts/en/2016-xx-xx-helpful-console-log-hacks.md]: same here

**Comment by gromgit** [comment]
:shipit:

**Comment by sjfkai** [comment]
What about adding [this link](https://developer.chrome.com/devtools/docs/javascript-debugging#breakpoints) to explain what is or how to set  a `conditional breakpoint`?

**Comment by gromgit** [comment]
@sjfkai Good point. The equivalent Firefox page is [here](https://developer.mozilla.org/en-US/docs/Tools/Debugger/How_to/Set_a_conditional_breakpoint).

**Comment by zackhall** [comment]
@sjfkai @gromgit great suggestion. I'll do that shortly. And here's the equivalent [Edge page](https://dev.windows.com/en-us/microsoft-edge/platform/documentation/f12-devtools-guide/debugger/#setting-and-managing-breakpoints) as well. 

@loverajoel that's awesome, I didn't know that was an option. Will add that as well.

**Comment by homer0** [comment]
Out of curiosity, why would you do `console.log(funcVariable + '');` instead of `console.log(funcVariable.toString())`?

**Comment by kurtextrem** [comment]
Shorter. Plus this is known and common practice in a lot of languages.

**Comment by adaniloff** [comment]
:shipit:

File: _posts/en/2016-xx-xx-helpful-console-log-hacks.md (hunk 1 above)
Comment (zackhall) [inline _posts/en/2016-xx-xx-helpful-console-log-hacks.md]: Done! Thanks for the suggestion!

**Comment by zackhall** [comment]
@sjfkai @gromgit Done! Thank you for the suggestions!

**Comment by homer0** [comment]
@kurtextrem I don't like mixing data types :P, but I get your point.

:shipit:

**Comment by zackhall** [comment]
yay

**Comment by zenopopovici** [comment]
I like your gifs guys ... here's mine :shipit: 
citizen kane - clapping

**Comment by kurtextrem** [comment]
I guess that is what we call "Tips after-launch party" :P
